tests/infra/member.py

```diff
@@ -111,7 +111,11 @@ def vote(
             and wait_for_global_commit
         ):
             with remote_node.node_client() as mc:
-                infra.checker.wait_for_global_commit(mc, r.seqno, r.view, True)
+                # If we vote in a new node, which becomes part of quorum, the transaction
+                # can only commit after it has successfully joined and caught up.
+                # Given that the retry timer on join RPC is 4 seconds, anything less is very
+                # likely to time out!
+                infra.checker.wait_for_global_commit(mc, r.seqno, r.view, True, timeout=6)
```

Comment:
Since this path is used for every proposal, would it be more precise to only set this timeout for the `trust_node` proposal?

Reply:
Yes, but it also seems like additional complexity. I think the comment explains the intent and will hopefully let us remove this altogether once join is fast enough, and in the meantime, it's just a slightly longer timeout for votes which shouldn't fail that often if at all in the CI (there's no penalty when they pass).